crates/ironrdp-rdpdr/src/pdu/esc.rs

```diff
@@ -931,12 +931,12 @@ pub mod rpce {
     ///
     /// Implementers should typically avoid implementing this trait directly
     /// and instead implement [`HeaderlessEncode`], and wrap it in a [`Pdu`].
-    pub trait Encode: PduEncode + Send + std::fmt::Debug {}
+    pub trait Encode: PduEncode + Send + Sync + std::fmt::Debug {}
 
     /// Trait for types that can be encoded into an [MS-RPCE] message.
     ///
     /// Implementers should typically implement this trait instead of [`Encode`].
-    pub trait HeaderlessEncode: Send + std::fmt::Debug {
+    pub trait HeaderlessEncode: Send + Sync + std::fmt::Debug {
```

Comment:
You’re used to it now, but here is the customary question (mostly for my own understanding): why do we need this bound on `Encode` and `HeaderlessEncode`?

Is `Sync` required for `DeviceControlResponse` (the struct holding a `Box<dyn rpce::Encode>`)?

If this is absolutely required, I suggest adding this line below `DeviceControlResponse`:
```rust
// Small comment explaining the rationale
ironrdp_pdu::assert_impl!(Config: Send, Sync);
```

Reply:
I'm not quite sure what happened, but it looks like you reviewed something from before I rebased. This is fixed now.

Reply:
Oh, all good then!